airbyte-integrations/connectors/source-greenhouse/source_greenhouse/greenhouse.yaml

```diff
@@ -0,0 +1,404 @@
+          path: "job_stages"
+  jobs_stages_stream:
+    type: DeclarativeStream
+    $options:
+      name: "jobs_stages"
+      primary_key: "id"
+      schema_loader:
+        $ref: "*ref(definitions.schema_loader)"
+      retriever:
+        $ref: "*ref(definitions.retriever)"
+        requester:
+          $ref: "*ref(definitions.requester)"
+          path: "jobs/{{ stream_slice.parent_id }}/stages"
+        stream_slicer:
+          type: SubstreamSlicer
+          parent_stream_configs:
+            - stream: "*ref(definitions.jobs_stream)"
+              parent_key: "id"
+              stream_slice_field: "parent_id"
+  offers_stream:
+    type: DeclarativeStream
+    $options:
+      name: "offers"
+      primary_key: "id"
+      schema_loader:
+        $ref: "*ref(definitions.schema_loader)"
+      retriever:
+        $ref: "*ref(definitions.retriever)"
+        requester:
+          $ref: "*ref(definitions.requester)"
+          path: "offers"
+  rejection_reasons_stream:
+    type: DeclarativeStream
+    $options:
+      name: "rejection_reasons"
+      primary_key: "id"
+      schema_loader:
+        $ref: "*ref(definitions.schema_loader)"
+      retriever:
+        $ref: "*ref(definitions.retriever)"
+        requester:
+          $ref: "*ref(definitions.requester)"
+          path: "rejection_reasons"
+  scorecards_stream:
+    type: DeclarativeStream
+    $options:
+      name: "scorecards"
+      primary_key: "id"
+      schema_loader:
+        $ref: "*ref(definitions.schema_loader)"
+      retriever:
+        $ref: "*ref(definitions.retriever)"
+        requester:
+          $ref: "*ref(definitions.requester)"
+          path: "scorecards"
+  sources_stream:
+    type: DeclarativeStream
+    $options:
+      name: "sources"
+      primary_key: "id"
+      schema_loader:
+        $ref: "*ref(definitions.schema_loader)"
+      retriever:
+        $ref: "*ref(definitions.retriever)"
+        requester:
+          $ref: "*ref(definitions.requester)"
+          path: "sources"
+  users_stream:
+    type: DeclarativeStream
+    $options:
+      name: "users"
+      primary_key: "id"
+      schema_loader:
+        $ref: "*ref(definitions.schema_loader)"
+      retriever:
+        $ref: "*ref(definitions.retriever)"
+        requester:
+          $ref: "*ref(definitions.requester)"
+          path: "users"
+  demographics_answers_stream:
+    type: DeclarativeStream
+    $options:
+      name: "demographics_answers"
+      primary_key: "id"
+      schema_loader:
+        $ref: "*ref(definitions.schema_loader)"
+      retriever:
+        $ref: "*ref(definitions.retriever)"
+        requester:
+          $ref: "*ref(definitions.requester)"
+          path: "demographics/answers"
+  demographics_answer_options_stream:
+    type: DeclarativeStream
+    $options:
+      name: "demographics_answer_options"
+      primary_key: "id"
+      schema_loader:
+        $ref: "*ref(definitions.schema_loader)"
```

Comment:
it's interesting how verbose / repititive this defintiion is for every stream. It would be really nice if we can DRY this by having a base stream extended by all of these e.g:

```
base_stream: 
  type: DeclarativeStream
  $options: 
    primary_key: "id"
    schema_loader: 
      $ref: "*ref(definitions.schema_loader)"
    retriever: 
      $ref: ""
       .....
```
and then each stream just configures the specific option they need

Reply:
yes!
I DRYed them config